analyzer/evmcontractcode/evm_contract_code.go

```diff
@@ -0,0 +1,198 @@
+package evmcontractcode
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/oasisprotocol/oasis-indexer/analyzer"
+	"github.com/oasisprotocol/oasis-indexer/analyzer/queries"
+	"github.com/oasisprotocol/oasis-indexer/analyzer/util"
+	"github.com/oasisprotocol/oasis-indexer/common"
+	"github.com/oasisprotocol/oasis-indexer/log"
+	"github.com/oasisprotocol/oasis-indexer/storage"
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi"
+)
+
+// This analyzer checks the list of addresses with an unknown is_contract status,
+// and determines it by calling `getCode()` on the address.
+// If the code is returned, it is also stored in the DB.
+// Every address that is the recipient of a call is a potential contract address.
+// Candidate addresses are inserted into the DB by the block analyzer.
+// Each candidate address only needs to be checked once.
+
+const (
+	EvmContractCodeAnalyzerPrefix = "evm_contract_code_"
+	MaxDownloadBatch              = 20
+	DownloadTimeout               = 61 * time.Second
+)
+
+type EthAddress []byte
+type OasisAddress string
+
+type Main struct {
+	runtime common.Runtime
+	source  nodeapi.RuntimeApiLite
+	target  storage.TargetStorage
+	logger  *log.Logger
+}
+
+var _ analyzer.Analyzer = (*Main)(nil)
+
+func NewMain(
+	runtime common.Runtime,
+	sourceClient nodeapi.RuntimeApiLite,
+	target storage.TargetStorage,
+	logger *log.Logger,
+) (*Main, error) {
```

Comment:
Nit: If this is changed to return the `Analyzer` interface `Main` can be unexported as well.